src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,61 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private Node<T> top;
+    private int size;
+
+    public static class Node<T> {
+        private Node<T> prev;
+        private T value;
 
+        public Node() {
+        }
+
+        public Node(T value, Node prev) {
+            this.prev = prev;
+            this.value = value;
+        }
+    }
+
+    public void push(T value) {
+        if (size == 0) {
+            Node<T> node = top;
+            top = new Node<>();
+            top.value = value;
+            top.prev = node;
+            size++;
+            return;
+        }
+        Node<T> newNode = new Node<>();
+        newNode.value = value;
+        newNode.prev = top;
+        top = newNode;
+        size++;
```

Comment:
I think it doesn`t matter what size this Stack have. To initialize fields of Node object you can use constructor. 
```suggestion
        top = new Node<>(value, top);
        size++;
```